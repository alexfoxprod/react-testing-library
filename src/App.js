import "./App.css";
import React, { useState } from "react";

const Search = ({ value, onChange, children }) => {
  <div>
    <label htmlFor="search">{children}</label>
    <input type="text" id="search" value={value} onChange={onChange} />
  </div>;
};
function App() {
  const [search, setSearch] = useState("");

  const handleChange = ({ target }) => {
    setSearch(target.value);
  };
  return (
    <div>
      <Search value={search} onChange={handleChange}>
        Search:
      </Search>
      <p>Searches for {search ? search : "..."}</p>
    </div>
  );
}

export default App;
