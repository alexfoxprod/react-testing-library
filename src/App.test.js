import { render, screen } from "@testing-library/react";
import App from "./App";

test("renders learn react link", () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
