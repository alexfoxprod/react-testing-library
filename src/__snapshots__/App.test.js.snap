// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div>
    <p>
      Searches for ...
    </p>
  </div>
</DocumentFragment>
`;
